packages/fluentui/react-northstar/src/utils/positioner/usePopper.ts

```diff
@@ -0,0 +1,391 @@
+  const arrowRef = useCallbackRef<HTMLElement | null>(null, handlePopperUpdate, true);
+
+  React.useImperativeHandle(
+    options.popperRef,
+    () => ({
+      updatePosition: () => {
+        popperInstanceRef.current?.update();
+      },
+    }),
+    [],
+  );
+
+  useIsomorphicLayoutEffect(() => {
+    handlePopperUpdate();
+
+    return () => {
+      popperInstanceRef.current?.destroy();
+      popperInstanceRef.current = null;
+    };
+  }, [options.enabled]);
+  useIsomorphicLayoutEffect(() => {
+    if (!isFirstMount) {
+      popperInstanceRef.current?.setOptions(
+        resolvePopperOptions(targetRef.current, containerRef.current, arrowRef.current),
+      );
+    }
+  }, [resolvePopperOptions]);
+
+  if (process.env.NODE_ENV !== 'production') {
+    // This checked should run only in development mode
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    React.useEffect(() => {
+      if (containerRef.current) {
+        const contentNode = containerRef.current;
+        const treeWalker = contentNode.ownerDocument?.createTreeWalker(contentNode, NodeFilter.SHOW_ELEMENT, {
+          acceptNode: hasAutofocusFilter,
+        });
+
+        while (treeWalker.nextNode()) {
+          const node = treeWalker.currentNode;
+          // eslint-disable-next-line no-console
+          console.warn('<Popper>:', node);
+          // eslint-disable-next-line no-console
+          console.warn(
+            [
+              '<Popper>: ^ this node contains "autoFocus" prop on a React element. This can break the initial',
+              'positioning of an element and cause a window jump effect. This issue occurs because React polyfills',
+              '"autoFocus" behavior to solve inconsistencies between different browsers:',
+              'https://github.com/facebook/react/issues/11851#issuecomment-351787078',
+              '\n',
+              'However, ".focus()" in this case occurs before any other React effects will be executed',
+              '(React.useEffect(), componentDidMount(), etc.) and we can not prevent this behavior. If you really',
+              'want to use "autoFocus" please add "position: fixed" to styles of the element that is wrapped by',
+              '"Popper".',
+              `In general, it's not recommended to use "autoFocus" as it may break accessibility aspects:`,
+              'https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-autofocus.md',
+              '\n',
+              'We suggest to use the "trapFocus" prop on Fluent components or a catch "ref" and then use',
+              '"ref.current.focus" in React.useEffect():',
+              'https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element',
+            ].join(' '),
+          );
+        }
+      }
+      // We run this check once, no need to add deps here
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
This warning is only executed on mount. So if `usePopper` is initially called with `enabled=false` (like in `PopperExamplePositioning.tsx`), this warning check will never be executed. 
However this was a problem even before refactor. So I don't think it should block this awesome PR

Reply:
Is it possible to add `options.enabled` as a dep then ?

Reply:
In TMP they first render a small popup (enabled=true), then the small popup is filled with proper content and become bigger (enabled=true still). So this check will still not be run for the bigger popup, which may have autoFocus. 

IMO, the only way to make sure this check always run, is to remove dependency at all. But then this check will run on every update. So I think this is a problem not worth solving.

Reply:
Good catch, this warning should also handle `containerRef` updates, it seems that it should not be a hook at all...

I added a TODO there, let's follow up on it in a separate PR.